### PR TITLE
Issue #288: match jar file name for CheckJarTest

### LINKED ISSUE #288: CheckJarTest should not rely on manual version update
In https://github.com/muhlba91/sonar-checkstyle/blob/master/src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java#L29 we rely on checking whether the jar file with a certain version got created.

This needs to be manually edited every time. Change to either automated detection of the version vi a `pom.xml` or changing the method to retrieve any jar file matching the name. Usually, especially on CI, only one jar file should be existing and not more (as opposed to some manual local testing).

### PR DESCRIPTION
**Pull request by muhlba91**
fix #288

### CODE REVIEW

File: src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java [hunk 1]
```diff
@@ -19,22 +19,27 @@
...
-        final boolean snapshotExists = new File("target/checkstyle-sonar-plugin-"
-                                                + VERSION + "-SNAPSHOT.jar").exists();
-        final boolean releaseExists = new File("target/checkstyle-sonar-plugin-"
-                                               + VERSION + ".jar").exists();
-        assertTrue("Jar should exists",
-                   snapshotExists || releaseExists);
+    public void testJarPresence() throws IOException {
+        final List<Path> files = Files.find(Paths.get("target"), 1, (path, basicFileAttributes) -> {
+            return path.toString()
+                    .matches(".*checkstyle-sonar-plugin-\\d+\\.\\d+(-SNAPSHOT)*\\.jar");
```
Comment (rnveach) [inline src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java]: > (-SNAPSHOT)*

Shouldn't this be `(\\-SNAPSHOT)?` ?

Comment (muhlba91) [reply src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java]: `?` is a good catch!
As for escaping the `-` it should not need escaping due to the grouping. The test runs through with normal and snapshot jars, no matter if I escape or not.

File: src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java [hunk 2]
```diff
@@ -19,22 +19,27 @@
...
-        final boolean releaseExists = new File("target/checkstyle-sonar-plugin-"
-                                               + VERSION + ".jar").exists();
-        assertTrue("Jar should exists",
-                   snapshotExists || releaseExists);
+    public void testJarPresence() throws IOException {
+        final List<Path> files = Files.find(Paths.get("target"), 1, (path, basicFileAttributes) -> {
+            return path.toString()
+                    .matches(".*checkstyle-sonar-plugin-\\d+\\.\\d+(-SNAPSHOT)?\\.jar");
+        })
+                .collect(Collectors.toList());
```
Comment (romani) [inline src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java]: please move lambda out to have a variable for it.
it is very hard to read code like this.

**Comment by rnveach** [comment]
@muhlba91 CI is failing. Checkstyle errors.

**Pull request merged**